### PR TITLE
Use new pinecone embeddings

### PR DESCRIPTION
**Pull request by mruwnik**
Handles both the old and new pinecone, as there are slight differences in dates and authors. Will require the appropriate env variables to be set

### CODE REVIEW

File: api/src/stampy_chat/env.py [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
+### OpenAI ###
+OPENAI_API_KEY   = os.environ.get('OPENAI_API_KEY')
 openai.api_key = OPENAI_API_KEY # non-optional
 
+### Pinecone ###
+PINECONE_API_KEY     = os.environ.get('PINECONE_API_KEY')
+PINECONE_ENVIRONMENT = os.environ.get('PINECONE_ENVIRONMENT', "us-east1-gcp")
+PINECONE_INDEX_NAME  = os.environ.get("PINECONE_INDEX_NAME", "alignment-search")
+PINECONE_INDEX       = None
+PINECONE_NAMESPACE   = os.environ.get("PINECONE_NAMESPACE", "alignment-search")  # "normal" or "finetuned" for the new index, "alignment-search" for the old one
```
Comment (mruwnik) [inline api/src/stampy_chat/env.py:26]: @tayler6000 once this gets merged, the following will have to be updated in the env:
```
PINECONE_API_KEY="5...<API key>...4"
PINECONE_INDEX_NAME="stampy-chat-ard"
PINECONE_ENVIRONMENT="us-central1-gcp"
PINECONE_NAMESPACE="normal"
```

**Review by ccstan99 — COMMENTED** [review]
LGTM but it would be better to have someone actually run it just to be sure everything works.

On the front-end, we should remove "using data through June 2022" so it just says "WARNING: This is a very early prototype. Feedback welcomed."

Fixes #82

**Comment by henri123lemoine** [comment]
Just tested it. It works great.

File: api/src/stampy_chat/get_blocks.py [hunk 2]
```diff
@@ -88,28 +90,38 @@ def get_top_k_blocks(index, user_query: str, k: int) -> List[Block]:
     blocks = []
     for match in query_response['matches']:
 
-        date = match['metadata']['date']
+        date = match['metadata']['date_published']
+
+        if isinstance(date, datetime.date):
+            date = date.isoformat()
+        elif isinstance(date, datetime.datetime):
+            date = date.date().isoformat
```
Comment (Thomas-Lemoine) [inline api/src/stampy_chat/get_blocks.py:98]: Is this not meant to be `date = date.date().isoformat()`?

Comment (henri123lemoine) [reply api/src/stampy_chat/get_blocks.py]: good point

**Review by Thomas-Lemoine — APPROVED** [review]

**Review by ccstan99 — APPROVED** [review]
I did a quick test and it worked for me too! However, I noticed a few things but we can deal these as separate issues:
- Blank PINECONE isn't working any more. Should go to chat.stampy.ai:PORT not railway, which doesn't work since pinecone format is mismatched
- Citations aren't listing authors only titles & dates
- Citations aren't getting deduplicated (example below)
- Even though temperature=0, the answers are not deterministic

Duplicated citations [1][3] and [2][4] when asked "What is AI Safety?"
[1] What is AI safety? - 2023-08-23
[2] What is the difference between AI safety, AI alignment, AI control, friendly AI, AI ethics, AI existential safety, and AGI safety? - 2023-08-19
[3] What is AI safety? - 2023-08-23
[4] Building safe artificial intelligence: specification, robustness, and assurance - 2018-09-26
[5] What is the difference between AI safety, AI alignment, AI control, friendly AI, AI ethics, AI existential safety, and AGI safety? - 2023-08-19

**Review by ccstan99 — APPROVED** [review]

**Pull request merged**